src/java.base/share/classes/java/lang/Integer.java

```diff
@@ -1767,6 +1767,102 @@ public static int reverse(int i) {
         return reverseBytes(i);
     }
 
+    /**
+     * Returns the value obtained by compressing the bits of the
+     * specified {@code int} value, {@code i}, in accordance with
+     * the specified bit mask.
+     * <p>
+     * For each one-bit value of the mask, {@code mb} say, from least
+     * significant to most significant, the bit value of {@code i} at
+     * the same bit location as {@code mb} is assigned to the compressed
+     * value contiguously starting from the least significant bit location.
+     * All the upper remaining bits of the compressed value are set
+     * to zero.
+     *
```

Comment:
Following Alan's comment, I suggest the following examples:

```
  compress(0x9ABCDEF, 0x0F0F0FF) == 0xACEF
  compress(x, 1<<n) == (x>>n & 1)
  compress(x, -1<<n) == x >>> n
  compress(m, m) == (m==-1||m==0)? m : (1<<bitCount(m))-1
  compress(x, m) == compress(x & m, m)
  compress(expand(x, m), m) == x & compress(m, m)
  (compress(x, m) >>> n) & 1 == /*the bit of x corresponding to the nth set bit in m*/
```

…In two places.  Similarly, examples for expand:

```
  expand(0x9ABCDEF, 0x0F0F0FF) == 0xC0D0EF
  expand(x, 1<<n) == (x&1) << n
  expand(x, -1<<n) == x << n
  expand(-1, m) == m
  expand(x, m) == expand(x, m) & m
  expand(compress(x, m), m) == x & m
  expand(1<<n, m) == /*the nth set bit in m, as a mask in place; cf. highest/lowestOneBit*/
```

(Please double check these examples!)

Reply:
Examples added in latest commit.

Reply:
I see you've added a comment on each example too, I think this really helpers readers to see how they can be used.

The class description of both Integer and Long have an implementation note (they pre-date @implNote) referencing Hacker's Delight. We could potentially expand the list of methods mentioned but it's not strictly necessary are they are just examples.

Reply:
I also saw that on the class doc and considered the new methods fit neatly under the category of "bit twiddling" methods.